From af26c4d4ab545767456d8c21ed48e9e01ce6a3e7 Mon Sep 17 00:00:00 2001
From: Yuxuan Shui <yshuiv7@gmail.com>
Date: Sat, 13 Oct 2012 18:44:15 -0700
Subject: [PATCH] Linux 3.6 compat, sops->write_super() removed

The .write_super callback was removed the the super_operations
structure by Linux commit f0cd2dbb6cf387c11f87265462e370bb5469299e.
All file systems are now expected to self manage writing any dirty
state assoicated with their super block.

ZFS never made use of this callback so it can simply be removed
from the super_operations structure.

Signed-off-by: Yuxuan Shui <yshuiv7@gmail.com>
Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
Issue #873
---
 module/zfs/zpl_super.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/module/zfs/zpl_super.c b/module/zfs/zpl_super.c
index 29d7f7d..fd4f691 100644
--- a/module/zfs/zpl_super.c
+++ b/module/zfs/zpl_super.c
@@ -316,7 +316,6 @@
 	.delete_inode		= zpl_inode_delete,
 #endif /* HAVE_EVICT_INODE */
 	.put_super		= zpl_put_super,
-	.write_super		= NULL,
 	.sync_fs		= zpl_sync_fs,
 	.statfs			= zpl_statfs,
 	.remount_fs		= zpl_remount_fs,
-- 
1.7.10

